objstore.go

```diff
@@ -829,3 +829,13 @@ func (t *timingReaderWriterTo) WriteTo(w io.Writer) (n int64, err error) {
 	t.timingReader.updateMetrics(int(n), err)
 	return n, err
 }
+
+type ObjectSizerReadCloser struct {
+	io.ReadCloser
+	Size func() (int64, error)
```

Comment:
From what I see, we always return `nil` as the error, so I wonder why we need to have it as a return argument.

Reply:
swift could potentially return err if the HEAD call fails